### PR TITLE
library using

### LINKED ISSUE #8: Support libraries
`library` doesn't have an analogue in Ligo. So it functions should be directly added to the contract.

### PR DESCRIPTION
**Pull request by vird**
* library using support
 * --collect_fn_decl, now only fn_decl which is more correct
 * fix bad opt.contract warning message
 * closes https://github.com/madfish-solutions/sol2ligo/issues/8 with some name collision issues
 * fix default_type_map_gen types

### CODE REVIEW

File: test/translate_ligo_library.coffee [hunk 1]
```diff
@@ -186,4 +186,158 @@ describe "translate ligo section library", ()->
...
+    type setBitExample_args is unit;
+    type state is record
+      const_ONE : nat;
+      const_ONES : nat;
+    end;
+    
+    function bits_setBit (const self : state; const tmp_self : nat; const index : nat) : (nat) is
+      block {
+        skip
+      } with (bitwise_or(tmp_self, bitwise_lsl(self.const_ONE, index)));
```
Comment (konchunas) [inline test/translate_ligo_library.coffee:322]: There are no longer (?) `bitwise_lsl` instruction.
It should be `Bitwise.shift_left` but that is left for a separate issue
https://ligolang.org/docs/reference/bitwise-reference/

File: test/translate_ligo_library.coffee [hunk 2]
```diff
@@ -186,4 +186,158 @@ describe "translate ligo section library", ()->
...
+        uint n = 0;
+        n = n.setBit(0); // Set the 0th bit.
+        assert(n == 1);  // 1
+      }
+    }
+    """
+    text_o = """
+    type setBitExample_args is unit;
+    type state is record
+      const_ONE : nat;
```
Comment (konchunas) [inline test/translate_ligo_library.coffee:315]: I would suggest translating `constant internal` as global contract-scope constants, not as state variables.
But that is unrelated issue though.

File: test/translate_ligo_library.coffee (hunk 1 above)
Comment (vird) [reply test/translate_ligo_library.coffee:322]: https://github.com/madfish-solutions/sol2ligo/issues/230
